### ISSUE TITLE
React-native Integration JS API

### ISSUE THREAD

**Issue by Profit-tech** [tradingview/charting-library-examples#248]
- I followed the example for React-native and it works for me.

<img width="363" alt="Screenshot 2021-09-28 at 11 11 45" src="https://user-images.githubusercontent.com/89030942/135049999-2d1635e8-bcdf-4f15-aa04-06fbe30c5590.png">

- But I cannot find a solution how can I integrate the js api.
Because if you follow the example. 
Included is an index.html file that is not at the level of my react-native application.
And I cannot dynamically pass data to the trade chart.

<img width="771" alt="Screenshot at Sep 28 11-20-49" src="https://user-images.githubusercontent.com/89030942/135051086-1d63e958-3a97-4da1-aaa0-7ae70b0313f3.png">

- If I include a library at the level of my application.

<img width="1042" alt="Screenshot at Sep 28 11-38-52" src="https://user-images.githubusercontent.com/89030942/135053653-3db60dad-bcff-4f9f-ab87-b4aafbd74b17.png">

- I am getting page loading error.

<img width="368" alt="Screenshot 2021-09-28 at 11 36 40" src="https://user-images.githubusercontent.com/89030942/135053234-9f80a050-8108-4af9-a1d3-305f0254d493.png">

- I have successfully connected JS API in web version of React-js.
   But it doesn't work in React-native for me.
- I am looking for any solutions, how can I integrate JS API and dynamically transfer data to a trade chart.

**Comment by timocov** [comment]
> That’s the reason why it’s impossible to send the actual data to the trade chart.

> Thus I'm looking for any solutions for JS API integration with an ability to send data to a trading chart dynamically.

I don't think that it is impossible because of that. Did you try to implement datafeed so it will request a data from your app rather that requesting it from the network. I think this should work fine.

**Comment by Profit-tech** [comment]
The question is exactly how to implement datafeed in react-native application.

Let’s follow through the example:
- Open Example.xcodeproj in Xcode.
- Make a right click on AppName and select "Add Files to "AppName"".
- Copy all files from https://github.com/tradingview/charting_library/.

As a result added files will include index.html (it describes in tradingview configuration).
It displays via the WebView in my application also (your could find it at the screen above).

So. My base question is how to change the data for datafeed in dynamic (relying on user’s actions) if there is no access to index.html file from my React-native application.

**Comment by makedonsky94** [comment]
Hi! Please, check the documentation of [React Native WebView](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Guide.md#communicating-between-js-and-native)

In the [example](https://github.com/tradingview/charting-library-examples/blob/master/react-native/app/ios.app.js) you can see Web-to-Native communication. 
1. Inject javascript 

    ```javascript
    injectedJavaScript={this.jsToInject}
    ```
    
    jsToInject
    
    ```javascript
    tvWidget.onChartReady(function() {
          tvWidget.chart().onIntervalChanged().subscribe(
              null,
              function(interval) {
                  const response = { type: "onIntervalChanged", interval: interval }
                  window.ReactNativeWebView.postMessage(JSON.stringify(response));
              }
          );
    });
    ```
    
2. Write a handler for that action
    ```javascript
    onMessage={(event) => {
        const data = JSON.parse(event.nativeEvent.data)
        if (data.type == "onIntervalChanged") {
            Alert.alert(
              'onIntervalChanged',
              "Interval = " + data.interval,
              [
                { text: 'OK', onPress: () => console.log('OK Pressed') }
              ],
              { cancelable: true }
            );
        }
    }}
    ```
    
    
 Now if you want to make Native-to-Web communication you should use `injectJavaScript` method
 
 ```javascript
 import React, { Component } from 'react';
import { View } from 'react-native';
import { WebView } from 'react-native-webview';

export default class App extends Component {
  render() {
    const run = `
      document.body.style.backgroundColor = 'blue';
      true;
    `;

    setTimeout(() => {
      this.webref.injectJavaScript(run);
    }, 3000);

    return (
      <View style={{ flex: 1 }}>
        <WebView
          ref={(r) => (this.webref = r)}
          source={{
            uri: 'https://github.com/react-native-webview/react-native-webview',
          }}
        />
      </View>
    );
  }
}
 ```

**Comment by Shai-E** [comment]
Did anyone come up with a working solution? Is there anyone who successfully integrated the charts in react native and made it look good on mobile?

**Comment by ronbansal** [comment]
Any updates on this? I am stuck in the same position

**Comment by Xplosive06** [comment]
The same here, don't know what to do...

**Comment by wakie92** [comment]
Same here......

**Comment by DavitVosk** [comment]
Almost 4 years and no update 🙈. Did someone find a solution please?

**Comment by VictorAO-django** [comment]
I'm in the same shoes right now 
Client thinks I'm unprofessional
